viadot/sources/redshift_spectrum.py

```diff
@@ -0,0 +1,199 @@
+        if not self._session:
+            self._session = boto3.session.Session(
+                profile_name=self.credentials["profile_name"],
+                aws_access_key_id=self.credentials["aws_access_key_id"],
+                aws_secret_access_key=self.credentials["aws_secret_access_key"],
+            )
+        return self._session
+
+    def ls(
+        self,
+        database: str = None,
+        name_contains: str = None,
+        search_text: str = None,
+    ) -> List[str]:
+        """
+        Returns a list of tables in a Glue database.
+
+        Args:
+            database (str, optional): Database name.
+            name_contains (str, optional): Match by a specific substring of the table
+                name.
+            search_text (str, optional): Select only tables with the given string in
+                table's properties.
+        """
+        df = wr.catalog.tables(
+            boto3_session=self.session,
+            database=database,
+            name_contains=name_contains,
+            search_text=search_text,
+        )
+
+        return df
+
+    def exists(self, database: str, table: str) -> bool:
+        """
+        Check if a table exists in Glue database.
+
+        Args:
+            database (str): AWS Glue catalog database name.
+            table (str): AWS Glue catalog table name.
+
+        Returns:
+            bool: Whether the paths exists.
+        """
+        return wr.catalog.does_table_exist(
+            boto3_session=self.session,
+            database=database,
+            table=table,
+        )
+
+    def rm(
+        self,
+        database: str,
+        table: str,
+        remove_files: bool = True,
+    ) -> None:
+        """
+        Deletes table from AWS Glue database and related file from AWS S3, if specified.
+
+        Args:
+            database (str): AWS Glue catalog database name.
+            table (str): AWS Glue catalog table name.
+            remove_files (bool, optional): If True, AWS S3 file related to the table
+                will be removed. Defaults to True.
+        """
+        if remove_files:
+            table_location = wr.catalog.get_table_location(
+                boto3_session=self.session,
+                database=database,
+                table=table,
+            )
+            wr.s3.delete_objects(path=table_location)
+
+        wr.catalog.delete_table_if_exists(database=database, table=table)
+
+    def from_df(
+        self,
+        df: pd.DataFrame,
+        to_path: str,
+        database: str,
+        table: str,
+        extension: str = ".parquet",
+        if_exists: Literal["overwrite", "append"] = "overwrite",
+        partition_cols: List[str] = None,
+        index: bool = False,
+        compression: str = None,
+        sep: str = ",",
+        description: str = "test",
+    ) -> None:
+        """
+        Upload a pandas `DataFrame` to a csv or parquet file.
```

Comment:
1. This function creates a Glue table, not just CSV/Parquet files. 
2. If it's supposed to be used to create a Spectrum table, it assumes an external schema was already created:
```python
create external schema "{raw_schema_name}" from data catalog 
database "{raw_schema_name}"
iam_role '{redshift_iam_role}'
region '{region_name}'
CREATE EXTERNAL DATABASE IF NOT EXISTS;
```
3. It's not possible to see from the code how can one use this Spectrum table (`SELECT * FROM ?`), probably an example would be useful here (esp. since it depends on how the schema was named when first created -- see point 2.)